src/modules/godot.rs

```diff
@@ -0,0 +1,169 @@
+use crate::configs::godot::GodotConfig;
+
+use super::{Context, Module};
+
+/// Creates a module that tells if the current directory is in a Godot project.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    use super::ModuleConfig;
+    use crate::formatter::StringFormatter;
+    use crate::formatter::VersionFormatter;
+
+    let godot_project_root = context
+        .begin_ancestor_scan()
+        .set_files(&["project.godot"])
+        .scan();
+    godot_project_root?;
```

Comment:
So far, we have only really used this for version control systems, I would prefer to keep this with the regular scanner and the related options.

Reply:
Is the ancestor scan reserved for VCS due to performance reasons? If so, would it be better to just have this module disabled by default?

I'd prefer to keep the ancestor scan for a couple reasons:

1. One of the common godot file extensions, `.gd`, is apparently shared by another project called [GD Graphics Library](https://libgd.github.io/), so searching for this extension could lead to false positives.
2. Personally, I like to know that I'm within a godot project folder even if I'm currently in an organizational subdirectory that doesn't contain any godot-specific file formats.

Reply:
If the module is disabled by default, adding `gd` should be less of an issue.

Reply:
If the module was disabled by default, wouldn't the ancestor scan also be less of an issue?